EngineLayer/GlycoSearch/Glycan.cs

```diff
@@ -0,0 +1,582 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Chemistry;
+using System;
+using Proteomics;
+using MassSpectrometry;
+
+namespace EngineLayer
+{
+    public class GlycanIon
+    {
+        public GlycanIon(string ionStruct, int ionMass, byte[] ionKind, int lossIonMass)
+        {
+            IonStruct = ionStruct;
+            IonMass = ionMass;
+            IonKind = ionKind;
+            LossIonMass = lossIonMass;
+        }
+        public string IonStruct { get; set; }
+        public int IonMass { get; set; }
+        public int LossIonMass { get; set; }//Glycan.Mass - IonMass
+        public byte[] IonKind { get; set; }
+    }
+
+    public class Glycan
+    {
+        public Glycan(string struc, int mass, byte[] kind, List<GlycanIon> ions, bool decoy)
+        {
+            Struc = struc;
+            Mass = mass;
+            Kind = kind;
+            Ions = ions;
+            Decoy = decoy;
+        }
+
+        public Glycan(byte[] kind)
+        {
+            Kind = kind;
+            Mass = GetMass(kind);
+        }
+
+        public int GlyId { get; set; }
+        public string Struc { get; private set; }
+        public int Mass { get; private set; }
+        public byte[] Kind { get; private set; }
+        public string Composition
+        {
+            get
+            {
+                return Glycan.GetKindString(Kind);
+            }
+        }
+        public List<GlycanIon> Ions { get; set; }
+        public bool Decoy { get; private set; }
+
+        public HashSet<int> DiagnosticIons
+        {
+            get
+            {
+                HashSet<int> diagnosticIons = new HashSet<int>();
+                if (Kind[0] >= 1)
+                {
+                    diagnosticIons.Add(10902895 - hydrogenAtomMonoisotopicMass);
```

Comment:
Kind[0], etc. same deal with the "magic numbers". i don't know what kind is and i don't know the difference between Kind[0], Kind[1], etc.

Reply:
Glycans are composed of several different types of mono saccharides and the number of each are correspond to Kind. 
I prefer to use magic number here for speed reason. I've try different ways and this is by far the best. 
The whole glycan is handled with Kind and the magic number are well defined. I will comment it well.